content/blog/2018/06/2018-06-08-jenkins-java10-hackathon.adoc

```diff
@@ -0,0 +1,82 @@
+---
+:layout: post
+:title: "Jenkins & Java 10+ Online Hackathon (Jun 18-22)"
+:tags:
+- events
+- community
+- developer
+:author: oleg_nenashev
+---
+
+image:/images/logos/formal_java/256.png[Jenkins Java, role=center, float=right]
+
+On behalf of the Jenkins Events Team,
+I would like to invite you to the “Jenkins & Java 10 Online Hackathon” which will take place from June 18th to 22nd.
+We will be working together on Jenkins core and plugins in order
+to find and fix compatibility issues, share experience and have some fun.
+Everybody is welcome to join, independently of their Jenkins experience and amount of time they have available.
+
+If you are interested in participating in the hackathon, please sign-up in
+link:https://docs.google.com/forms/d/1ReYyuyCGC0PIz2quh6XehnjpH2K52inx-veHLPlNreE/edit[this form].
+
+### Background
+
+Java 9 has recently been end-of-lifed, Java 10 in GA, and Java 11 is in early beta.
+Jenkins project currently requires Java 8 to run reliably,
+and there are some known compatibility issues with higher Java versions.
+
+During the Jenkins World 2017 Hackathon,
+link:https://github.com/markewaite[Mark Waite] and
+link:https://github.com/batmat[Baptiste Mathus] spent some time exploring Java 9 compatibility in Jenkins.
+We are currently tracking compatibility issues in the
+link:https://issues.jenkins-ci.org/browse/JENKINS-40689[JENKINS-40689] EPIC,
+but there are likely many unknown issues in Jenkins core, plugins and in libraries we use in the project.
+We would like to continue their effort and work on Java 10+ support.
+
+### Objectives and Scope
+
+As I have said above,
+the goals are to explore/fix compatibility issues, share experience and have fun.
+We **DO NOT** plan to make Jenkins fully compatible with Java 10+
+during the hackathon,
+but we will try to integrate fixes and make them available.
+
+Since the announcement of the Hackathon in the mailing list,
+we have got a number of registrations from contributors working on several project areas.
+We will split our work to several areas:
+
+* Jenkins core and Remoting
+* Pipeline Engine
+* Plugins (e.g. Git plugin or any plugin you want to work on)
+* Exploratory testing for Java 10 and beyond
+
+In order to organize the effort, we have created a
+link:https://issues.jenkins-ci.org/issues/?jql=labels%20%3D%20java10_hackathon[java10_hackathon] label
```

Comment:
Seems clearer to use an epic.

Reply:
EPIC contents are not visible to users who are not logged into the system.
Also, plugins and components may have different EPICs for it

Reply:
> EPIC contents are not visible to users who are not logged into the system.

Surely participants will need a JIRA login at a minimum?!

> plugins and components may have different EPICs for it

Why…?

Reply:
I will reconsider the approach if we get too many contributors, JIRA/documentation updates are on me in this case

> Surely participants will need a JIRA login at a minimum?!

Maybe not. I barely can imagine how they can proceed without it, but some bits can be done (e.g. Docker packaging testing, JCasC, whatever other things with GitHub trackers)

> > plugins and components may have different EPICs for it
> Why…?

E.g. Pipeline support may be a separate EPIC due to its specifics.
Maven plugin should be a separate story, because I do not want to have it a s a blocker for Java 10 support by Jenkins.